openqa-post-similarity

```diff
@@ -0,0 +1,126 @@
+import json
+import argparse
+import logging
+
+"""
+This script scans all autoinst-log.txt files in the testresults directory searching
+for error messages. Then it computes the Levenshtein distance and posts the
+results as openQA comments.
+"""
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+help_message = "This script scans all autoinst-log.txt files in the testresults directory searching for error messages. Then it " \
+    "computes the Levenshtein distance and posts the results as openQA comments. "
+parser = argparse.ArgumentParser(description=help_message)
+args = parser.parse_args()
+
+# gather all the autoinst-log.txt
+logger.info("Get environment variables")
+BASE_DIR = os.getenv("OPENQA_BASEDIR")
+if BASE_DIR == None:
+    BASE_DIR = "/var/lib/"
+
+
+# I used a JSON file to store some intermediate computation results to reduce time cost.
+# This part finds error messages in autoinst-log.txt
+# Maybe there are multiple errors in a single autoinst-log.txt
+# I think some different code may trigger the same bug.
+# The first line is the most important, so I focus on the first lines of the errors.
+if not os.path.exists("id_msg.json"):
+    logger.info("id_msg.json does not exist.")
+    logs = []
+    for root, dirs, files in os.walk(BASE_DIR + "/openqa/testresults"):
+        for name in files:
+            if name == "autoinst-log.txt":
+                logs.append(os.path.join(root, name))
+    id_msg = {}
+    for path in logs:
+        with open(path, 'r') as f:
+            text = f.read()
+            anchor = 0
+            begin = text.find("Test died", anchor)
+            while begin != -1:
+                anchor = begin + 1
+                end = text.find("\n", begin)
+                id_msg[path.split("\\")[1]] = text[begin:end]
+                begin = text.find("Test died", anchor)
+    id_msg_json = json.dumps(id_msg, sort_keys=False, indent=4, separators=(',', ':'))
+    with open("id_msg.json", "w") as f:
+        f.write(id_msg_json)
+
+# If there exists the JSON file
+else:
+    logger.info("id_msg.json exists.")
+    with open("id_msg.json", "r") as f:
+        id_msg = json.load(f)
+    all_dirs = os.listdir(BASE_DIR + "/openqa/testresults")
+    checked_dirs = list(id_msg.keys())
+    unchecked_dirs = [i for i in all_dirs if i not in checked_dirs]
+    for job_id in unchecked_dirs:
+        path = BASE_DIR + "/openqa//testresults/" + job_id + "/autoinst-log.txt"
+        with open(path, 'r') as f:
+            text = f.read()
+            anchor = 0
+            begin = text.find("Test died", anchor)
+            while begin != -1:
+                anchor = begin + 1
+                end = text.find("\n", begin)
+                id_msg[path.split("\\")[1]] = text[begin:end]
+                begin = text.find("Test died", anchor)
+    id_msg_json = json.dumps(id_msg, sort_keys=False, indent=4, separators=(',', ':'))
+    with open("id_msg.json", "w") as f:
+        f.write(id_msg_json)
```

Comment:
This code looks like a copy of the above code, so it should be extracted to a function.

Reply:
I missed your comments before. Thank you.